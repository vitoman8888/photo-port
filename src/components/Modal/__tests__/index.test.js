import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Modal from '../'

afterEach(cleanup)

const otherCurrentPhoto = {
    name: 'portraits', 
    description: 'Portraits of people in my life', 
    category: 'landscape', 
    index: 1
};
const mockOnClose = jest.fn();

describe('Modal component', () => {
    it('renders', () => {
        render(<Modal onClose={mockOnClose} currentPhoto={otherCurrentPhoto} />);
    })

      // snapshot test
      it('matches snapshot DOM node structure', () => {
        const { asFragment } = render(
            <Modal onClose={mockOnClose} 
                    currentPhoto={otherCurrentPhoto} />);
        expect(asFragment()).toMatchSnapshot();
    });
});

describe('Click Event', () => {
    it('calls onClose handler', () => {
      const { getByText } = render(<Modal
        onClose={mockOnClose}
        currentPhoto={otherCurrentPhoto}
      />);
      fireEvent.click(getByText('Close this modal'));
  
      expect(mockOnClose).toHaveBeenCalledTimes(1);
    });
});  