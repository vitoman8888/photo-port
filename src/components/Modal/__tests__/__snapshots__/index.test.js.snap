// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div
    class="modalBackdrop"
  >
    <div
      class="modalContainer"
    >
      <h3
        class="modalTitle"
      >
        portraits
      </h3>
      <img
        alt="category"
      />
      <p>
        Portraits of people in my life
      </p>
      <button
        type="button"
      >
        Close this modal
      </button>
    </div>
  </div>
</DocumentFragment>
`;
